elastic/datadog_checks/elastic/elastic.py

```diff
@@ -439,6 +461,123 @@ def _process_cat_allocation_data(self, admin_forwarder, version, base_tags):
                 desc = CAT_ALLOCATION_METRICS[metric]
                 self._process_metric(cat_allocation_dic, metric, *desc, tags=tags)
 
+    def _process_custom_metric(
+        self, value, data_path, value_path, dynamic_tags, xtype, metric_name, tags=None, hostname=None
+    ):
+        """
+        value: JSON payload to traverse
+        data_path: path to data right before metric value or right before list of metric values
+        value_path: path to data after data_path to metric value
+        dynamic_tags: list of tags and their value_paths
+        xtype: datadog type
+        metric_name: datadog metric name
+        tags: a lambda to apply to the numerical value
+        hostname: a lambda to apply to the numerical value
+        """
+
+        tags_to_submit = deepcopy(tags)
+        path = data_path + '.' + value_path
+
+        # Collect the value of tags first, and then append to tags_to_submit
+        for (dynamic_tag_path, dynamic_tag_name) in dynamic_tags:
+            # Start at where data_path left off
+            dynamic_tag_value = value
+
+            # Traverse down the tree to find the tag value
+            for key in re.split(REGEX, dynamic_tag_path):
+                if dynamic_tag_value is not None:
+                    dynamic_tag_value = dynamic_tag_value.get(key.replace('\\', ''))
+                else:
+                    self.log.debug("Dynamic tag not found: %s -> %s", path, dynamic_tag_name)
+                    break
+
+            # If tag is there, then add it to list of tags to submit
+            if dynamic_tag_value is not None:
+                dynamic_tag = '{}:{}'.format(dynamic_tag_name, dynamic_tag_value)
+                tags_to_submit.append(dynamic_tag)
+            else:
+                self.log.debug("Dynamic tag is null: %s -> %s", path, dynamic_tag_name)
+
+        # Now do the same for the actual metric
+        branch_value = value
+        for key in re.split(REGEX, value_path):
+            if branch_value is not None:
+                branch_value = branch_value.get(key.replace('\\', ''))
+            else:
+                break
```

Comment:
It looks like this logic is repeated multiple times, shall we extract it to a function "get value using path" ?

https://github.com/DataDog/integrations-core/blob/a75a5fe1364bcc7c132f90417a95bdae5c6ca82e/elastic/datadog_checks/elastic/elastic.py#L548-L553

and other places where `REGEX` is used.